### PR TITLE
Add Chemdner dataset loader

### PR DESCRIPTION
**Pull request by qanastek**
Hello,

I finish the implementation of the CHEMDNER dataset and validate all the steps from the [CONTRIBUTING.md](https://github.com/bigscience-workshop/biomedical/blob/master/CONTRIBUTING.md).

The only point I cannot find online is the `_LICENSE` of the dataset. I saw the [Public w/ Registration](https://github.com/bigscience-workshop/biomedical/labels/Public%20w%2F%20Registration) label on the issue itself, but nothing elsewhere.

### CODE REVIEW

File: biodatasets/chemdner/chemdner.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and Samuele Garda.
```
Comment (galtay) [inline biodatasets/chemdner/chemdner.py:2]: sorry, this snuck in from a time when we were putting names in instead of using github to track contributions. can change this to the value in the template please? 
```
# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
```

**Review by galtay — CHANGES_REQUESTED** [review]
dataset loader script looks good! just a couple fixes outside the core logic

File: biodatasets/chemdner/chemdner.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+_DESCRIPTION = """\
+We present the CHEMDNER corpus, a collection of 10,000 PubMed abstracts that contain a total of 84,355 chemical entity
+mentions labeled manually by expert chemistry literature curators, following annotation guidelines specifically
+defined for this task. The abstracts of the CHEMDNER corpus were selected to be representative for all major chemical
+disciplines. Each of the chemical entity mentions was manually labeled according to its structure-associated chemical
+entity mention (SACEM) class: abbreviation, family, formula, identifier, multiple, systematic and trivial.
+"""
+
+_HOMEPAGE = "https://biocreative.bioinformatics.udel.edu/resources/biocreative-iv/chemdner-corpus/"
+_LICENSE = " CC0 1.0 Universal (CC0 1.0) Public Domain Dedication"
```
Comment (galtay) [inline biodatasets/chemdner/chemdner.py:126]: To discover the appropriate licensing information for this dataset, I would, 
* check for any information in a README or LICENSE file that comes with the dataset download
* if nothing is there, check the website for the dataset (https://biocreative.bioinformatics.udel.edu/resources/biocreative-iv/chemdner-corpus/)
* if nothing is there, check publications announcing the dataset
* if nothing is there, check the organization's website. 

Looking around I can't find anything specific to this dataset on the BioCreative website (please double check that). If nothing turns up, then I think our fall back for BioCreative should be to use 
`_LICENSE = https://biocreative.bioinformatics.udel.edu/accounts/register/`  ... this appears to have the most general statement of their data usage policy  ... Note that for some BioCreative datasets (e.g. BC5CDR) there is a more specific license included in the dataset download.

Comment (qanastek) [reply biodatasets/chemdner/chemdner.py]: https://zenodo.org/record/4957519#.Yka_iudBybg

File: tests/test_bigbio.py [hunk 3]
```diff
@@ -637,4 +637,4 @@ def _test_has_only_one_item(self, msg: str, field: list):
     TestDataLoader.DATA_DIR = args.data_dir
     TestDataLoader.USE_AUTH_TOKEN = args.use_auth_token
 
-    unittest.TextTestRunner().run(TestDataLoader())
+    unittest.TextTestRunner().run(TestDataLoader())
```
[the remark this answers is not in the document]
Comment (qanastek) [reply tests/test_bigbio.py]: I restore the file to the current master status using `git restore -s origin/master -- tests/test_bigbio.py`

**Comment by qanastek** [comment]
Commit fbaa0abe547318f62853cf654d6fb3d1c1c1efd2

**Comment by galtay** [comment]
It looks like changes to the test_bigbio.py file still exist in this PR. One way to remove these changes from your PR is to checkout the file from the upstream/master branch (for example https://itsnotbugitsfeature.com/2020/01/24/removing-a-specific-file-from-a-pull-request/) ... in this case it would mean doing something like, 
```
git checkout chemdner
git checkout upstream/master -- tests/test_bigbio.py
git commit -m 'removing edits to tests'
git push origin chemdner
```

File: biodatasets/chemdner/chemdner.py (hunk 2 above)
Comment (galtay) [reply biodatasets/chemdner/chemdner.py]: that's a nice find, but I think the license there applies to the annotation guide not the dataset.

**Comment by qanastek** [comment]
I have done this process again and the file is already up-to-date. At the current state, the file is identical to the `origin/master`. Only the commit tree contains the previous "edits".

**Comment by qanastek** [comment]
`git checkout chemdner`

> Already on 'chemdner'

`git checkout upstream/master -- tests/test_bigbio.py`

> fatal: invalid reference: upstream/master

**Comment by qanastek** [comment]
`git checkout origin/HEAD -- tests/test_bigbio.py`

`git add .`

`git commit -m "removing edits to tests"`

> On branch chemdner

> nothing to commit, working tree clean
